### PR TITLE
WIP: Make use of Podman-compose

### PR DESCRIPTION
**Pull request by jaybatra26**
Since Docker-compose is going to be decommissioned.
This PR replaces docker-compose with podman-compose.

Fixes: #2158
Signed-off-by: Jay Batra <jaybatra73@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@ishankhare07, would you like to review this PR?

**Review by leecalcote — CHANGES_REQUESTED** [review]
@jaybatra26, `docker-compose` will still be a supported platform for Meshery (the default platform). `podman-compose` will be an additionally supported platform. You'll want to add support for `podman-compose` to most of the `mesheryctl system` commands (start, stop, restart, update, logs, channel...) 

See the `platform` attribute in `~/.meshery/config.yaml` as the key to use when adding "podman" as an option. `system context` is a new comment, not yet released into a stable Meshery release. `system context` creates and controls `config.yaml` and currently only supports "docker" as a value. It needs to support "podman" and "kubernetes" as values, too.

**Comment by leecalcote** [comment]
Thank you for taking this on, @jaybatra26 👍

**Comment by jaybatra26** [comment]
@leecalcote  It's my pleasure. :)

**Comment by jaybatra26** [comment]
Hi @leecalcote Do I have to make changes in a this file most likely. https://github.com/layer5io/meshery/blob/master/mesheryctl/pkg/utils/helpers.go#L208

**Comment by tripathyprateek** [comment]
Yes we need to try and test how podman-compose behaves with Meshery. I guess it will be a good enough issue to tackle for some time. And we need to keep support for Docker as default platform. Also there seems some more development around Podman to support Docker-compose with Podman 3.0 [here](https://www.redhat.com/sysadmin/podman-docker-compose)

**Comment by anirudhjain75** [comment]
Hi @jaybatra26 , Do look into the golint error caused in the CI

**Comment by jaybatra26** [comment]
Thanks @tripathyprateek  and @anirudhjain75 sure.

File: mesheryctl/internal/cli/root/system/start.go [hunk 1]
```diff
@@ -96,159 +96,159 @@ func start() error {
 	if err != nil {
 		return errors.Wrap(err, "error processing config")
 	}
-	if(mctlCfg.Contexts["local"].Platform== "docker") 
-	{
+	ctxPlatform := mctlCfg.GetContextContent().Platform
+	if ctxPlatform == "docker" || ctxPlatform == "" {
```
Comment (anirudhjain75) [inline mesheryctl/internal/cli/root/system/start.go:100]: We might not want to default to docker. @leecalcote shed some light if possible. I dont think we should do anything or maybe give an error if platform is not set.

File: mesheryctl/internal/cli/root/system/start.go [hunk 2]
```diff
@@ -96,159 +96,159 @@ func start() error {
 	if err != nil {
 		return errors.Wrap(err, "error processing config")
 	}
-	if(mctlCfg.Contexts["local"].Platform== "docker") 
-	{
+	ctxPlatform := mctlCfg.GetContextContent().Platform
```
Comment (anirudhjain75) [inline mesheryctl/internal/cli/root/system/start.go:99]: These is an addon that can be done here. from #2256 I am planning to add a context flag. Taking that into consideration for which context to focus on would be great. Another issue could be created for this.

**Review by anirudhjain75 — COMMENTED** [review]
The changes look fine to be. ignoring the docker as default for now. @leecalcote might be able to shed some light

**Comment by leecalcote** [comment]
> Yes we need to try and test how podman-compose behaves with Meshery. I guess it will be a good enough issue to tackle for some time. And we need to keep support for Docker as default platform. Also there seems some more development around Podman to support Docker-compose with Podman 3.0 [here](https://www.redhat.com/sysadmin/podman-docker-compose)

@tripathyprateek I would love to see results from your compatibility tests here. 👍

**Comment by leecalcote** [comment]
@tripathyprateek have you been successful in running Meshery on Podman? Any luck progressing on the mesheryctl command here? 

Assuming we're in about the same spot, I'll suggest that we push this feature to post v0.5.0. What do you think?

**Comment by tripathyprateek** [comment]
> @tripathyprateek have you been successful in running Meshery on Podman? Any luck progressing on the mesheryctl command here?
> 
> Assuming we're in about the same spot, I'll suggest that we push this feature to post v0.5.0. What do you think?

@leecalcote I have been trying to run Meshery on Podman. However I am  running into a lot of errors with Podman and Buildah which I am trying to figure out. I would suggest we can wait to add support for Podman for a little while as Podman is not fully working on some platforms (Ubuntu 20.10 and earlier) I guess.

**Pull request closed without merging**